src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-tpm2.c

```diff
@@ -77,6 +68,14 @@ _public_ int cryptsetup_token_open(
         assert(token == r);
         assert(json);
 
+        /* pin was passed as pin = pin, pin_size = strlen(pin). We need to add terminating
+         * NULL byte to addressable memory */
+        if (pin && pin[pin_size-1] != '\0') {
+                pin_string = strndup(pin, pin_size);
+                if (!pin_string)
+                        return crypt_log_oom(cd);
+        }
```

Comment:
i figure we should refuse pins with embedded NUL bytes.

Reply:
also, please usue memdup_suffix0

Reply:
> i figure we should refuse pins with embedded NUL bytes.

Indeed, this code is copied from https://github.com/systemd/systemd/blob/14e7bc2e77d2699498a1f74d7e4f905f11eca335/src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c#L46-L55 and I forgot to copy the relevant `memchr` part :see_no_evil: Fixed now, thanks!

> also, please usue memdup_suffix0

Done, I also fixed https://github.com/systemd/systemd/blob/14e7bc2e77d2699498a1f74d7e4f905f11eca335/src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c#L52 in a separate commit.